src/controller/http.rs

```diff
@@ -0,0 +1,82 @@
+use std::{sync::Arc, time::Duration};
+
+use tokio::time::MissedTickBehavior;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    gateway_allocations::allocations_checker::NoopAllocationsChecker, storage::manager::Status,
+};
+
+use super::worker::Worker;
+
+const PING_TIMEOUT: Duration = Duration::from_millis(200);
+
+pub struct HttpController {
+    worker: Arc<Worker<NoopAllocationsChecker>>,
```

Comment:
Why is the worker wrapped in `Arc` and its fields too? This looks like to many `Arc`s to me. We should remove either this or the inner `Arc`s.

Reply:
You're right, the `Worker` takes ownership of everything passed to it. Thanks!